File: 1-js/1-js-basics/3.2-conditional-statements-advanced-exercise/04-fishingBoat.js
function solve([budget, season, num]) {

    budget = Number(budget);

    num = Number(num);

    let total = 0;

    let data = {

        'Spring': 3000,

        'Summer': 4200,

        'Autumn': 4200,

        'Winter': 2600

    }

    total += data[season];

    if (num <= 6) {

        total *= 0.90;

    } else if (num > 6 && num <= 11) {

        total *= 0.85;

    } else {

        total *= 0.75;

    }

    if (num % 2 == 0 && season != 'Autumn') {

        total *= 0.95;

    }

    return total <= budget ? `Yes! You have ${(budget - total).toFixed(2)} leva left.` : `Not enough money! You need ${(total - budget).toFixed(2)} leva.`;

}

console.log(solve(['2000', 'Winter', '13']));

File: 2-python/1-python-basics/3.2-conditional-statements-advanced-exercise/04-fishing_boat.py
budget = int(input())
season = input()
fishermen_count = int(input())
ship_price_spring = 3000
ship_price_summer_autumn = 4200
ship_price_winter = 2600
price = 0

if season == 'Spring':
    price = ship_price_spring
elif season == 'Summer' or season == 'Autumn':
    price = ship_price_summer_autumn
else:
    price = ship_price_winter

if fishermen_count <= 6:
    price *= 0.9
elif 7 < fishermen_count <= 11:
    price *= 0.85
else:
    price *= 0.75

if fishermen_count % 2 == 0 and season != 'Autumn':
    price *= 0.95

result = budget - price

if result >= 0:
    print(f'Yes! You have {result:.2f} leva left.')
else:
    print(f'Not enough money! You need {abs(result):.2f} leva.')
